lib/discover/local.go

```diff
@@ -281,3 +288,30 @@ func (c *localClient) registerDevice(src net.Addr, device Announce) bool {
 
 	return isNewDevice
 }
+
+// filterUnspecifiedLocal returns the list of addresses after removing any
+// unspecified, localhost, multicast, broadcast or port-zero addresses.
+func filterUnspecifiedLocal(addrs []string) []string {
+	filtered := addrs[:0]
+	for _, addr := range addrs {
+		u, err := url.Parse(addr)
+		if err != nil {
+			continue
+		}
+
+		tcpAddr, err := net.ResolveTCPAddr("tcp", u.Host)
+		if err != nil {
+			continue
+		}
```

Comment:
Do we want to filter out quic addresses here? There's probably no scenario where local discovery works but local tcp doesn't, and udp does, but as long as it's not definitely impossible (I don't know enough to say that) it wouldn't hurt to send them anyway.

Reply:
We don't filter out quic. The resolve says "tcp", but I don't think that matters.

Reply:
My bad, I assumed this would always error for quic-schemed addresses, but `u.Host` is just `ip:port` at this point. Anyway indeed hardly relevant.